pkg/straceback/straceback.go

```diff
@@ -221,11 +224,12 @@ func (sb *StraceBack) DumpProg(id uint32) (out string, err error) {
 		return "", fmt.Errorf("invalid index")
 	}
 
-	_ = sb.tracelets[id].tailCallProg.PerfMapStop("events")
-	/* ignore error: it might have been stopped already */
-
-	arr := sb.tracelets[id].pm.DumpBackward()
-	out = eventsToString(eventsToGo(arr))
+	arr := sb.tracelets[id].pm.SwapAndDumpBackward()
+	sb.tracelets[id].eventBuffer = append(sb.tracelets[id].eventBuffer, eventsToGo(arr)...)
```

Comment:
I cannot understand how it could work. wouldn't it print the same event multiple times to the terminal?
Maybe I am missing something obvious.

Reply:
The screen is cleared. I should add a comment on that ;)

Reply:
In Inspektor Gadget the `show` command always shows the full log and the fact that we have to swap the buffers is abstracted away.

Reply:
Added the comment about the screen clearing.

Reply:
Ok perfect, it looks for me now then.